src/main/java/seedu/address/model/AlfredModel.java

```diff
@@ -0,0 +1,108 @@
+package seedu.address.model;
+
+import java.nio.file.Path;
+import java.util.function.Predicate;
+
+import javafx.collections.ObservableList;
+import javafx.collections.transformation.FilteredList;
+import seedu.address.commons.core.GuiSettings;
+import seedu.address.commons.exceptions.AlfredException;
+import seedu.address.model.entity.Id;
+import seedu.address.model.entity.Mentor;
+import seedu.address.model.entity.Participant;
+import seedu.address.model.entity.Team;
+import seedu.address.model.entitylist.ReadOnlyEntityList;
+import seedu.address.model.person.Person;
+import seedu.address.storage.AlfredStorage;
+
+/**
+ * The API of the AlfredModel component.
+ */
+public interface AlfredModel {
+    /**
+     * {@code Predicate} that always evaluate to true
+     */
+    Predicate<Person> PREDICATE_SHOW_ALL_PERSONS = unused -> true;
+
+    /**
+     * Initializes the alfredModel.
+     */
+
+    /**
+     * Replaces user prefs data with the data in {@code userPrefs}.
+     */
+    void setUserPrefs(ReadOnlyUserPrefs userPrefs);
+
+    /**
+     * Returns the user prefs.
+     */
+    ReadOnlyUserPrefs getUserPrefs();
+
+    /**
+     * Returns the user prefs' GUI settings.
+     */
+    GuiSettings getGuiSettings();
+
+    /**
+     * Sets the user prefs' GUI settings.
+     */
+    void setGuiSettings(GuiSettings guiSettings);
+
+    /**
+     * Returns the ParticipantList.
+     */
+    FilteredList<Participant> getParticipantFilteredList();
+
+    /**
+     * Returns the TeamList.
+     */
+    FilteredList<Team> getTeamFilteredList();
+
+    /**
+     * Returns the MentorList.
+     */
+    FilteredList<Mentor> getMentorFilteredList();
+
+    /* Get the filtered lists */
+
+    /* Below is the API exposed for the controllers to call */
+
+    /* Participant methods */
+
+    Participant getParticipant(Id id) throws AlfredException;
+
+    void addParticipant(Participant participant) throws AlfredException;
+
+    void updateParticipant(Id id, Participant participant) throws AlfredException;
+
+    Participant deleteParticipant(Id id) throws AlfredException;
+
+    /* Team methods */
+
+    Team getTeam(Id teamId) throws AlfredException;
+
+    Team getTeamByParticipantId(Id participantId) throws AlfredException;
+
+    Team getTeamByMentorId(Id mentorId) throws AlfredException;
+
+    void addTeam(Team team) throws AlfredException;
+
+    void addParticipantToTeam(Id teamId, Participant participant) throws AlfredException;
+
+    void addMentorToTeam(Id teamId, Mentor mentor) throws AlfredException;
+
+    void updateTeam(Id teamId, Team team) throws AlfredException;
+
+    Team deleteTeam(Id id) throws AlfredException;
+
+    /* Mentor methods */
+
+    Mentor getMentor(Id id) throws AlfredException;
+
+    void addMentor(Mentor mentor) throws AlfredException;
+
+    void updateMentor(Id id, Mentor mentor) throws AlfredException;
+
+    Mentor deleteMentor(Id id) throws AlfredException;
+
```

Comment:
Hey :) I think AlfredModel might need to be updated with a few methods. When I implemented the Model History object, I made changes to AlfredModel to include a updateHistory method as well. I'm re-looking at whether or not this needs to be re-implemented, but in any case, based on the current implementation in master, I think there'll be a compile error for this, so just a heads up!